### DISCUSSION TITLE
When a ListItem is Selected how do I return the Label text?

### DISCUSSION THREAD

**Discussion by showipintbri** [Textualize/textual#1840] [Q&A] [answered]
Using the example ListView widget code: [https://github.com/Textualize/textual/blob/main/docs/examples/widgets/list_view.py](https://github.com/Textualize/textual/blob/main/docs/examples/widgets/list_view.py)

...and this reference in the documentation [https://textual.textualize.io/api/list_view/#textual.widgets._list_view.ListView.Selected](https://textual.textualize.io/api/list_view/#textual.widgets._list_view.ListView.Selected)

I should be able to catch the selected object:
> Can be handled using `on_list_view_selected` in a subclass of ListView or in a parent widget in the DOM.

However, when I create a `on_list_view_selected` function I don't see the Label text passed. How can I access the Label text?

**Comment by davep** [comment, marked as answer]
A `ListItem` is a container widget, that can contain other widgets; in the example you point at it will contain a `Label`; it could, if you wanted, be any other kind of widget. The point here being that a `ListItem` doesn't have a label; it's just a container of other things.

If you want the `ListItem`s of your `ListView` to contain widgets that also let you get a string label back, likely the best approach is to make such a widget of your own. So, taking the `list_view.py` example a little further:

```python
from textual.app import App, ComposeResult
from textual.widgets import ListView, ListItem, Label, Footer

class LabelItem(ListItem):

    def __init__(self, label: str) -> None:
        super().__init__()
        self.label = label

    def compose( self ) -> ComposeResult:
        yield Label(self.label)


class ListViewExample(App):

    CSS_PATH = "list_view.css"

    def compose(self) -> ComposeResult:
        yield ListView(
            LabelItem("One"),
            LabelItem("Two"),
            LabelItem("Three"),
        )
        yield Label("Nothing chosen",id="chosen")
        yield Footer()

    def on_list_view_selected(self, event: ListView.Selected):
        self.query_one("#chosen",Label).update(event.item.label)


if __name__ == "__main__":
    app = ListViewExample()
    app.run()
```

**Comment by showipintbri** [comment]
@davep Thank You. Your suggestion achieves the outcome I was hoping for. 

While I have adopted your suggested code, I still seek clarity on how to access the text of a child widget.

In the example: 
```
yield ListView(
            ListItem(Label("One")),
            ListItem(Label("Two")),
            ListItem(Label("Three")),
        )
```

Is there another way to access the text of Label when it's a child of a child using the  `on_list_view_selected(self, event: ListView.Selected):` handler ?

**Reply by davep** [reply]
> Is there another way to access the text of Label 

A `Label` doesn't really have "text" as such, a `Label` is a way of showing a Rich renderable, which may come from some text.

**Comment by stragulus** [comment]
Since this is such a common use case (wanting to get an object associated with a selected item), it would make sense to add this as standard functionality to ListItem to avoid having to always create a subclass.

**Comment by daxroc** [comment]
You can do something like 
```
datasource = ListView(...)
ch = datasource.highlighted_child.get_child_by_type(Label)
ds_text = ch.renderable.plain
```

**Comment by travis-aaron** [comment]
I ended up using Static for this

```python
yield ListItem(Static("My text", name="My text"))
```
You can get the text from `name`

```python
def on_list_view_selected(self, event: ListView.Selected) -> None:
    list_item = event.item.children[0]
    self.notify(f"{list_item.name}")
```

**Comment by gflcampos** [comment]
The easiest way is to query the ListItem for the Label:
```py
yield ListItem(Label("My item"), id="my-item")
```

```py
def on_list_view_selected(self, event: ListView.Selected) -> None:
    selected_label = event.item.query_one(Label).renderable
```

**Comment by galaktycznyfunk** [comment]
All of this seems to be a rather convoluted way of avoiding `label.text` attribute. Wouldn't that just significantly simplify things?
